src/main/java/org/example/hw5/src/main/java/org/example/repository/impl/RoleRepositoryImpl.java

```diff
@@ -0,0 +1,34 @@
+package org.example.repository.impl;
+
+import org.example.model.Role;
+import org.example.repository.RoleRepository;
+import org.example.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+public class RoleRepositoryImpl implements RoleRepository {
+    private final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Role create(Role role) {
+        try (Session session = sessionFactory.openSession()) {
+            session.save(role);
+        }
+        return role;
+    }
+
+    @Override
+    public Optional<Role> findById(Long id) {
+        return Optional.ofNullable(sessionFactory.openSession().find(Role.class, id));
```

Comment:
it's better to use try-with-resourses construction and sessions will be closed automatically